langkit/documentation.py

```diff
@@ -196,6 +196,10 @@ def instantiate_templates(doc_dict):
         Nevertheless, the library does its best not to crash the program,
         materializing internal errors using this kind of exception.
     """,
+    'langkit.precondition_failure': """
+        Exception raised when user-input passed to a langkit-provided API did
+        not satisfy the specification of that API.
```

Comment:
I suggest we avoid referring to Langkit in generated APIs (i.e. when talking to the users of generated libraries). What about:
```
Exception raised when an API is called while its preconditions are not satisfied.
```